src/main/java/gov/nasa/jpf/constraints/expressions/PropositionalCompound.java

```diff
@@ -93,10 +93,19 @@ public LogicalOperator getOperator() {
 
   @Override
   public void print(Appendable a, int flags) throws IOException {
+    if(left == null){
+      System.out.println(operator.toString());
+      right.print(System.out);
+      System.out.flush();
+    }
     a.append('(');
     left.print(a, flags);
     a.append(' ').append(operator.toString()).append(' ');
-    right.print(a, flags);
```

Comment:
You are right that a malformed expression should yield an exception during usage, but I used the print function for trace based debugging, because some arithmetic done on a malformed expression yields an exception. In this case it is not helpful if a print statement also yields an exception.
To solve this, my proposal is to reestablish old print behavior and add printMalformedExpression to jConstraints as done in the new commits. So this method could be used for debugging and the normal print could be used for stable products.
It might be more a work setup issue, but I would request to support trace based debugging in jConstraints.